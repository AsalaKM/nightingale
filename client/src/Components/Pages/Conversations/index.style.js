
import styled from 'styled-components'


export const Div = styled.div`
    text-align: center;

`


export const ViewDiv = styled.div`
    padding:1px;

`
export const Items = styled.div`
    display:  flex;
    width: 50%;
    margin: 0 auto;
    border-top: 1px solid #ccc;
    height: 5rem;
    align-items: center; 
    background: #FAFAFA;
    justify-content: space-around;
    padding: 0 3rem 0 3rem;
    cursor:pointer;

        @media (max-width: 768px) {
            width: 100%;
            padding: 0;
            border-top: none;
            border-bottom: 1px solid #ccc;
        }
`

export const ErrorMsg = styled.p`
    color:#444;
    font-size:1.5rem;
`
export const Item = styled.p`
text-decoration: none;
:nth-child(1) {
    font-size:2rem;
  }
:nth-child(2) {
    color: black !important;
    font-weight: bold;
  }
  :nth-child(3) {
    color: #8C8C8C !important;
  }
  :nth-child(4) {
    display: none;
  }

`