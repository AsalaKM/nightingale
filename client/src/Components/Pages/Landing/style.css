.landing{
    background: linear-gradient(to top, #925DE5, #cec8d8, #fff);
    height: 100vh;
    text-align: center;
}