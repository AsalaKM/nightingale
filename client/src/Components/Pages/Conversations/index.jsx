import React, { Component } from "react";
import axios from "axios";
import uuid from "uuid";
import { Div, ViewDiv, ErrorMsg, Items, Item } from "./index.style";
import Title from "../../Common/Title";

class Conversations extends Component {
  state = {
    conversations: [],
    message: ""
  };

  getConversations = async () => {
    const { id } = this.props.match.params;
    const data = await axios(`/api/user/${id}/conversations`);
    const finalData = data.data;
    const { conversations } = this.state;
    if (finalData.length === 0) {
      const msg = " There is no conversations yet !!";
      this.setState({ message: msg, conversations: [] });
    } else {
      finalData.map(row =>
        conversations.push([
          row.mood[0].moodEmoji,
          row.dayOfWeek,
          row.date,
          row._id
        ])
      );
      this.setState({ conversations });
    }
  };

  componentDidMount = () => {
    this.getConversations();
  };
  onClick = conversation => {
    const user = this.props.match.params.id;
    this.props.history.push(`/${user}/conversations/${conversation}`);
  };

  render() {
    const { conversations, message } = this.state;
    return (
      <>
        <Title value="Conversations" />

        <Div>
          {message && (
            <ErrorMsg>
              {message} <i className="far fa-surprise" />
            </ErrorMsg>
          )}
          {conversations.map(row => (
            <ViewDiv onClick={() => this.onClick(row[3])}>
              <Items key={uuid()}>
                {row.map(value => (
                  <Item key={uuid()}>{value}</Item>
                ))}
              </Items>
            </ViewDiv>
          ))}
        </Div>
      </>
    );
  }
}

export default Conversations;
