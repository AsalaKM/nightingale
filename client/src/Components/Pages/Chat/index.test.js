import React from "react";
import renderer from "react-test-renderer";
import Conversations from "./index.jsx";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { ReplyButton } from "./index.style";

// set enzyme configuration
configure({ adapter: new Adapter() });

describe("Chat", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Conversations />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("mounts correctly", done => {
    const wrapper = mount(<Conversations />);
    expect(1).toBe(1);
    // mock the data
    const state = {
      botCardReply: null,
      botMessage: null,
      botQuickReply: ["Amazing", "Good2", "Meh", "Not great", "Terrible"],
      conversation: [
        { cardReply: null, quickReply: [], text: "Hi! ðŸ‘‹", user: "ai" },
        {
          cardReply: null,
          quickReply: [],
          text: "How was your school day?",
          user: "ai"
        },
        {
          cardReply: null,
          quickReply: ["Amazing", "Good", "Meh", "Not great", "Terrible"],
          text: "",
          user: "ai"
        },
        {
          userMessage: ""
        }
      ]
    };

    wrapper.setState(state);

    // it should have a 5 reply buttons
    expect(wrapper.find(`.${ReplyButton.styledComponentId}`)).toHaveLength(5);

    // expect the first reply card to have a "Amazing" value
    const amazingButton = wrapper
      .find(`.${ReplyButton.styledComponentId}`)
      .first();
    expect(amazingButton.html()).toMatch("Amazing");

    // state length before select a reply
    const oldState = wrapper.state().conversation.length;

    // simulate "click" event on reply buuton
    amazingButton.simulate("click");

    // the new state after selecting a reply
    const newState = wrapper.state().conversation.length;

    //the new reply should be stored in the state
    expect(newState - oldState).toBe(1);
    window.location = "/";
    done();
  });
});
