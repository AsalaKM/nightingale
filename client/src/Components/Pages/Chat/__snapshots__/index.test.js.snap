// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat renders correctly 1`] = `
<div>
  <div
    className="sc-bdVaJa dzaoIy"
  >
    <div
      className="sc-bwzfXH bHHejb"
    />
    <div
      className="sc-htpNat dJofZH"
    >
      <form
        className="sc-bxivhb hmqJnS"
        onSubmit={[Function]}
      >
        <input
          autoFocus={true}
          className="text-input"
          onChange={[Function]}
          placeholder="Type your message and hit enter to send"
          type="text"
          value=""
        />
      </form>
      <div
        style={
          Object {
            "clear": "both",
            "float": "left",
          }
        }
      />
    </div>
  </div>
</div>
`;
